src/js/actions/markActions.ts

```diff
@@ -36,6 +37,18 @@ export function addMark (record: MarkRecord) {
     }, id));
   };
 }
+
+export function addGroup(record: MarkRecord, layoutId: number, dir: string) {
+  return function(dispatch: Dispatch, getState: () => State) {
```

Comment:
i remember now: 
- `import {ThunkDispatch} from 'redux-thunk';`
- change the type annotation for `dispatch` from `Dispatch` to `ThunkDispatch<State, any, any>`
- remove the `as any` below